src/secure-postgresql-db/index.ts

```diff
@@ -0,0 +1,551 @@
+import * as cdk from '@aws-cdk/core';
+import { Duration, RemovalPolicy, Tags } from 'aws-cdk-lib';
+import { BackupPlan, BackupPlanRule, BackupPlanRuleProps, BackupResource } from 'aws-cdk-lib/aws-backup';
+import { Alarm, ComparisonOperator } from 'aws-cdk-lib/aws-cloudwatch';
+import { SnsAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
+import { ISecurityGroup, IVpc, InstanceType, Port, SecurityGroup, SubnetType, Vpc } from 'aws-cdk-lib/aws-ec2';
+import { Schedule } from 'aws-cdk-lib/aws-events';
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { RetentionDays } from 'aws-cdk-lib/aws-logs';
+import { CfnDBInstance, Credentials, DatabaseInstance, DatabaseInstanceEngine, ParameterGroup, PostgresEngineVersion } from 'aws-cdk-lib/aws-rds';
+import { Secret } from 'aws-cdk-lib/aws-secretsmanager';
+import { Topic } from 'aws-cdk-lib/aws-sns';
+import { Construct, IConstruct } from 'constructs';
+import _ from 'lodash';
+import { EzConstruct } from '../ez-construct';
+import { Defaults } from '../lib/defaults';
+import { Utils } from '../lib/utils';
+
+
+interface AlarmProps {
+  cpuUtilizationThresholdPct?: number;
+  minFreeMemoryMB?: number;
+  maxOpenConnections?: number;
+  freeStorageThresholdPct?: number;
+}
+
+
+interface SecurePostgresqlDbProps {
+  vpcId: string;
+  cloudWatchLogRetention?: RetentionDays;
+  postgresEngineVersion?: PostgresEngineVersion;
+  instanceIdentifier?: string | null;
+  instanceType: InstanceType;
+  port?: number;
+  overrideInstanceProps?: { [key: string]: any } | {};
+  additionalSecurityGroups?: ISecurityGroup[];
+  allocatedStorage?: number;
+  maxAllocatedStorage?: number;
+  multiAz?: boolean;
+  backupPlanRuleProps?: BackupPlanRuleProps;
+  alarmProps?: AlarmProps;
+  snapshotArn?: string | null;
+}
+
+
+export class SecurePostgresqlDb extends EzConstruct {
+  public encryptionKey: Key;
+  public instance: DatabaseInstance;
+  public credentialSecret: Secret;
+  public securityGroup: SecurityGroup;
+  public backupPlan: BackupPlan;
+  public alarms: Alarm[];
+
+  private _props: SecurePostgresqlDbProps;
+  private _vpc: IVpc;
+  private _alarmTopic: Topic;
+  private _allowedSgs: ISecurityGroup[] = [];
+  private _assembled = false;
+  private _instanceTags: [string, string][] = [];
+  private _constructTags: [string, string][] = [];
+
+  private _defaultProps: Defaults<SecurePostgresqlDbProps> = {
+    cloudWatchLogRetention: RetentionDays.THREE_MONTHS,
+    postgresEngineVersion: PostgresEngineVersion.VER_15_3,
+    instanceIdentifier: null,
+    port: 5432,
+    allocatedStorage: 50,
+    maxAllocatedStorage: 200,
+    multiAz: false,
+    overrideInstanceProps: { deletionProtection: false },
+    additionalSecurityGroups: [],
+    backupPlanRuleProps: {
```

Comment:
Do not enforce this in defaults, our idea it to utilize what is set globally at the organization level utilizing the tags. 
We should have a method to capture the backupPlanRuleProp insted. 

```ts 
 let db = new SecuePostgresqlDb()
 db.backupPlanRule(...)   // something like this

```

Reply:
Handled with `enableBackupPlan` which defaults to false